word_frequency.py

```diff
@@ -0,0 +1,56 @@
+banned_chars = [".", ",", '"', "'", "?", "!", "@", "&", "%",
+                "$", "' ", " - ", " '"]
+spaced_banned = ["' ", " - ", " '",]
+
+def clean_it(word):
+    """Strips unwanted characters from the file."""
+    banned_chars = [".", ",", '"', "'", "?", "!", "@", "&", "%",
+                    "$", "' ", " - ", " '"]
+    spaced_banned = ["' ", "-", " '", ' ']
+
+    word = word.lower()
+
+    for char in word:
+        if char in banned_chars:
+            word = word.replace(word, '')
+
+        elif char in spaced_banned:
+            word = word.replace(word, ' ')
+
+    return word
+
+
+
+def word_frequency(file_name):
+    "Determines how many times a word occurs in a text file."""
+
+    with open(file_name) as text:
+        file = text.readlines()
+        word_dict = {}
+        clean_list = []
+        final_list = []
+
+        for word in file:
+            word = word.split()
+            clean_list.extend(word)
+
+        for word in clean_list:
+            final_list.append(clean_it(word))
+
+        for word in final_list:
+            if word in word_dict:
+                word_dict[word] += 1
+            else:
+                word_dict[word] = 1
+    sorted_dict = sorted(word_dict.items(), key=lambda x: x[1], reverse=True)
+    top_20 = sorted_dict[1:20] # couldn't figure out how to leave spaces out of count, so I started at the next highest value
```

Comment:
Thanks for the advice, fixed the name of my variable, still working out the "do nothing" for spaces...

Sent from my iPhone

> On Jan 14, 2015, at 9:59 PM, "Clinton N. Dreisbach" <notifications@github.com> wrote:
> 
> In word_frequency.py:
> 
> > +        final_list = []
> > +
> > +        for word in file:
> > +            word = word.split()
> > +            clean_list.extend(word)
> > +
> > +        for word in clean_list:
> > +            final_list.append(clean_it(word))
> > +
> > +        for word in final_list:
> > +            if word in word_dict:
> > +                word_dict[word] += 1
> > +            else:
> > +                word_dict[word] = 1
> > +    sorted_dict = sorted(word_dict.items(), key=lambda x: x[1], reverse=True)
> > +    top_20 = sorted_dict[1:20] # couldn't figure out how to leave spaces out of count, so I started at the next highest value
> You could leave out spaces a couple of ways. I'd eliminate them in the if statement starting on line 41 by doing nothing if the word is made of spaces. You could also use a list comprehension in several places.
> 
> On line 45, you have the variable sorted_dict which isn't a dict. That's a big concern -- names are how we understand the code, and if they're not accurate, we easily misunderstand.
> 
> —
> Reply to this email directly or view it on GitHub.
>